File: week2/eunbeen/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<Integer> answer = new ArrayList();
        Queue<Integer> queue = new LinkedList();

        //작업에 필요한 일자 저장
        for(int i=0; i<progresses.length; i++){
            int day = (int)Math.ceil((double)(100-progresses[i])/speeds[i]);
            queue.add(day);
        }
        
        int count = 1;
        int release = queue.remove();
        
        while(queue.size() > 0){
            //같이 배포가 가능한 경우
            if(release >= queue.peek()){
                count += 1;
                queue.remove();
            }
            //같이 배포가 불가능한 경우
            else{
                answer.add(count);
                count = 1;
                release = queue.remove();
            }
        }
        
        if(count >= 1) answer.add(count);
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: week2/cylee/기능개발.py
import math
def get_dueto_day(current_percent, per_percent):
    return math.ceil((100-current_percent) / per_percent)

def solution(progresses, speeds):
    answer = []
    limit_days = []
    for i in range(len(speeds)):
        # 소요일수 구하고
        limit_days.append(get_dueto_day(progresses[i], speeds[i]))

    pre = limit_days[0]
    count = 1
    for idx, i in enumerate(limit_days):
        # 다음이 같거나 작으면 카운팅
        if(idx ==0): continue
        if(i<=pre): count += 1
        else:
            answer.append(count)
            count = 1
            pre = i
    answer.append(count)
    
        
    return answer

print(solution([98,99,98]	,[1, 1,1]		))